packages/mui-docs/src/ComponentLinkHeader/ComponentLinkHeader.tsx

```diff
@@ -107,14 +107,13 @@ export function ComponentLinkHeader(props: ComponentLinkHeaderProps) {
             size="small"
             variant="outlined"
             rel="nofollow"
-            href={`${process.env.SOURCE_CODE_REPO}/blob/v${process.env.LIB_VERSION}/${headers.githubSource}`}
+            href={`${process.env.SOURCE_CODE_REPO}/tree/v${process.env.LIB_VERSION}/${headers.githubSource}`}
```

Comment:
```suggestion
            href={`${process.env.SOURCE_CODE_REPO}/${/\.[a-z]{2,3}$/.test(headers.githubSource) ? 'blob' : 'tree'}/v${process.env.LIB_VERSION}/${headers.githubSource}`}
```
Maybe I'm overengineering But with a regexp that check if the file ends with a dot plus 2 or 3 characters, you get a fix that work for all cases.

Reply:
I would imagine that on the API page we always link files and on the demo page folders, so it might be fine?
If we start to see regressions, yeah, I would add an exception here, it must be a folder.

Reply:
This component is not on the API page. It's the header of the demo. Technically, we could put files or folders in the markdown header